mapper/orm/src/main/java/org/hibernate/search/mapper/orm/massindexing/impl/IdentifierConsumerDocumentProducer.java

```diff
@@ -190,37 +191,38 @@ private void rollbackTransaction(SessionImplementor session) throws Exception {
 		}
 	}
 
-	private void indexAllQueue(List<?> entities) throws InterruptedException, ExecutionException {
+	private void indexAllQueue(SessionImplementor session, List<?> entities) throws InterruptedException, ExecutionException {
 		if ( entities == null || entities.isEmpty() ) {
 			return;
 		}
-		else {
-			log.tracef( "received a list of objects to index: %s", entities );
-			for ( Object object : entities ) {
-				try {
-					index( object );
-					// TODO: implements monitor
-					// monitor.documentsBuilt( 1 );
-				}
-				catch (RuntimeException e) {
-					String errorMsg = log.massIndexerUnableToIndexInstance(
-							object.getClass().getName(),
-							object.toString()
-					);
-
-					// TODO: implements exception handler
-					// errorHandler.handleException( errorMsg, e );
-					// temporary re-throw the exception
-					throw new RuntimeException( errorMsg, e );
-				}
+
+		//TODO: switch to specific batch-optimized workPlan
+		PojoWorkPlan workPlan = mapping.createSearchManager( session ).createWorkPlan();
```

Comment:
No need to change it, since you will have to switch to a different kind of work plan later, but just for your information: work plans are reusable. You could have created it in the run() method and reused it for each `List` of entities. Though you may have wanted a `clear()` method to call in case something goes wrong.